providers/openfeature-flagd-provider/lib/openfeature/flagd/provider/client.rb

```diff
@@ -98,7 +98,7 @@ def prepare_evaluation_context(evaluation_context)
           return nil unless evaluation_context
 
           fields = evaluation_context.fields
-          fields["targetingKey"] = fields.delete(:targeting_key)
+          fields["targetingKey"] = fields.delete("targeting_key")
```

Comment:
EvaluationContext on initialization converts all keys to string, thats why it wasn't working before.